File: e-book-repository/src/main/webapp/services/categoryService.js
/**
 * Created by Jasmina on 19/06/2018.
 */
app.service('categoryService', function($http){
    return {
        read: function(onSuccess, onError){
            $http.get('/api/categories').then(onSuccess, onError);
        },
        create: function(category, onSuccess, onError){
            $http.post('/api/categories', category).then(onSuccess, onError);
        },
        update: function(category, onSuccess, onError){
            $http.patch('/api/categories/' + category.id, category).then(onSuccess, onError);
        },
        delete: function (id, onSuccess, onError) {
            $http.delete('/api/categories/' + id).then(onSuccess, onError);
        }
    }
});

File: e-book-repository/src/main/java/com/ftn/service/CategoryService.java
package com.ftn.service;

import com.ftn.model.dto.CategoryDTO;

import java.util.List;

/**
 * Created by Jasmina on 16/06/2018.
 */
public interface CategoryService {

    List<CategoryDTO> readAll();

    CategoryDTO create(CategoryDTO categoryDTO);

    CategoryDTO update(Long id, CategoryDTO categoryDTO);

    boolean delete(Long id);
}
